src/App/Fossa/Test.hs

```diff
@@ -0,0 +1,174 @@
+
+module App.Fossa.Test
+  ( testMain
+  ) where
+
+import Prologue
+import qualified Prelude as Unsafe
+
+import qualified App.Fossa.Analyze.FossaV1 as Fossa
+import App.Fossa.Analyze.ProjectInference
+import Control.Carrier.Error.Either
+import Control.Concurrent (threadDelay)
+import qualified Control.Concurrent.Async as Async
+import qualified Data.Map as M
+import qualified Data.Text as T
+import Data.Text.IO (hPutStrLn)
+import Effect.Logger
+import Network.HTTP.Req
+import Path.IO
+import System.IO (stderr)
+import System.Exit (exitSuccess, exitFailure)
+
+pollDelaySeconds :: Int
+pollDelaySeconds = 8
+
+testMain
+  :: Url 'Https -- ^ api base url
+  -> Text -- ^ api key
+  -> Severity
+  -> Int -- ^ timeout (seconds)
+  -> Maybe Text -- ^ cli override for name
+  -> Maybe Text -- ^ cli override for revision
+  -> IO ()
+testMain baseurl apiKey logSeverity timeoutSeconds overrideName overrideRevision= do
+  basedir <- getCurrentDir
+
+  void $ timeout timeoutSeconds $ withLogger logSeverity $ do
+    result <- runError @TestError $ do
+      inferred <- inferProject basedir
+      let revision = Fossa.ProjectRevision
+            (fromMaybe (inferredName inferred) overrideName)
+            (fromMaybe (inferredRevision inferred) overrideRevision)
+
+      logInfo ""
+      logInfo ("Using project name: `" <> pretty (Fossa.projectName revision) <> "`")
+      logInfo ("Using revision: `" <> pretty (Fossa.projectRevision revision) <> "`")
+
+      logSticky "[ Waiting for build completion... ]"
+
+      waitForBuild baseurl apiKey revision
+
+      logSticky "[ Waiting for issue scan completion... ]"
+      issues <- waitForIssues baseurl apiKey revision
+      logSticky ""
+
+      if null (Fossa.issuesIssues issues)
+        then logInfo "Test passed! 0 issues found"
+        else do
+          logError (renderedIssues issues)
+          liftIO exitFailure
+
+    case result of
+      Left err -> do
+        logError $ pretty (renderTestError err)
+        liftIO exitFailure
+      Right _ -> liftIO exitSuccess
+
+  -- we call exitSuccess/exitFailure in each branch above. the only way we get
+  -- here is if we time out
+  hPutStrLn stderr "Timed out while waiting for issues scan"
+  exitFailure
+
+renderedIssues :: Fossa.Issues -> Doc ann
+renderedIssues issues = rendered
```

Comment:
cliv1 used a template renderer here -- and we don't have something similar for haskell. Instead, we're using the `prettyprinter` library